### ISSUE TITLE
Can't set env with a comment

### ISSUE THREAD

**Issue by KolesnichenkoDS** [eslint/eslint#2705]
**Code**:
```javascript
/* eslint-env es6 */

[1, 2, 3, 4, 5].map((n) => {
    "use strict";
    return n * n;
});
```
**Error**:
```
script.js
  2:26  error  Unexpected token >

✖ 1 problem (1 error, 0 warnings)
```

But when I remove comment and add **package.json** it works well:
```json
{
    "eslintConfig": {
        "env": {
            "es6": true
        }
    }
}
```

**Comment by gcochard** [comment]
This is a parser flag, and must be set before scanning begins. Duplicate of #2614.

**Comment by ilyavolodin** [comment]
This is a known limitation. We haven't figured out a way to get it to work for now. We have other tickets tracking this issue, so closing as duplicate

**Comment by RReverser** [comment]
> We haven't figured out a way to get it to work for now.

Can't you reg-match corresponding comments in the input string before the scanning begins?

**Comment by ilyavolodin** [comment]
I think that's what we ended up doing. es6 environment should not work as comments.

**Comment by mysticatea** [comment]
#3314

**Comment by RReverser** [comment]
@mysticatea Ah, it uses different syntax than other options. Didn't see this in docs (is it reflected?). Thanks anyway!
